### PR TITLE
Notifications

### PR DESCRIPTION
**Pull request by ianballard**
Phase 1 - Simple ajax-notification base. Task Lists can be shared among users and users will be alerted of the newly available entity.

### CODE REVIEW

**Comment by ianballard** [comment]
Forgot to update notification bell image to use th:classappend. I'll update that

File: src/main/resources/db/migration/V3.0__task_list_users.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
+create table `task_list_users` (
```
Comment (jjameson25) [inline src/main/resources/db/migration/V3.0__task_list_users.sql:1]: Forgive my novice grasp of Kotlin/Thymeleaf. Why do you need this additional table? (trying to understand, not provide design input)

Comment (jjameson25) [reply src/main/resources/db/migration/V3.0__task_list_users.sql:1]: Ian has explained this to me in person. Thanks @iballard95

File: src/main/kotlin/com/jahnelgroup/domain/notification/NotificationRepo.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.jahnelgroup.domain.notification
+
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.Query
+
+interface NotificationRepo : JpaRepository<Notification, Long> {
+
+    @Query("select n from Notification n where n.recipient = ?#{ principal?.username } order by n.isRead, n.createdDatetime desc")
```
Comment (jjameson25) [inline src/main/kotlin/com/jahnelgroup/domain/notification/NotificationRepo.kt:8]: Don't forget your NOLOCKs!!

**Review by jjameson25 — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/jahnelgroup/domain/notification/NotificationRepo.kt (hunk 2 above)
Comment (jjameson25) [reply src/main/kotlin/com/jahnelgroup/domain/notification/NotificationRepo.kt:8]: Review comment aside... how does this change the functionality? If you left this line out would it break the application?

Comment (jjameson25) [reply src/main/kotlin/com/jahnelgroup/domain/notification/NotificationRepo.kt:8]: Oh wait, may have answered my own question here - this query is just for that particular function on line 9, right?

Comment (ianballard) [reply src/main/kotlin/com/jahnelgroup/domain/notification/NotificationRepo.kt:8]: correct. I am leaving out the no lock for now as I do not think we want dirty reads on this but down the line someone may want them.

**Pull request merged**